graphql_compiler/cost_estimation/analysis.py

```diff
@@ -0,0 +1,238 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+import bisect
+from dataclasses import dataclass
+from typing import Any, Dict, List
+
+from cached_property import cached_property
+
+from ..compiler.compiler_frontend import graphql_to_ir
+from ..compiler.metadata import FilterInfo, QueryMetadataTable
+from ..cost_estimation.cardinality_estimator import estimate_query_result_cardinality
+from ..cost_estimation.int_value_conversion import (
+    convert_int_to_field_value,
+    field_supports_range_reasoning,
+    is_uuid4_type,
+)
+from ..cost_estimation.interval import Interval
+from ..global_utils import PropertyPath, QueryStringWithParameters, VertexPath
+from ..schema import is_meta_field
+from ..schema.schema_info import QueryPlanningSchemaInfo
+from .filter_selectivity_utils import (
+    adjust_counts_for_filters,
+    get_integer_interval_for_filters_on_field,
+)
+
+
+def _convert_int_interval_to_field_value_interval(
+    schema_info: QueryPlanningSchemaInfo, vertex_type: str, field: str, interval: Interval[int]
+) -> Interval[Any]:
```

Comment:
Add docstring. This should probably give you a linting error.

Reply:
will add, but there's no linting error